### DISCUSSION TITLE
Can not get algorithm  aes-256-gcm+ed25519 for my license using java.

### DISCUSSION THREAD

**Discussion by felcodev** [keygen-sh/community#29] [General]
I have used the example for java license files but I can not get the aes-256-gcm+ed25519 algorithm within my license. I am generating the license file by using the GET method url web service. And I also get a base64+ed25519  algorithm not a aes-256-gcm+ed25519 as it shows in the example. The encodedIv also errors. 

Is there something that I am doing wrong? I am generating my license file correctly?

**Comment by ezekg** [comment]
You need to encrypt the license file by setting [the `encrypt` query parameter](https://keygen.sh/docs/api/licenses/#licenses-check-out-query-encrypt). By default, license files are not encrypted.

```
GET https://api.keygen.sh/v1/accounts/{ACCOUNT}/licenses/{ID}/actions/check-out?encrypt=true
```

**Comment by felcodev** [comment]
Ok I think I know what to do. Stand by. Lol

J.D. Felkins
FELCO Solutions, Inc.
www.felcosolutions.com<http://www.felcosolutions.com>
Email: ***@***.******@***.***>
Main: 770-766-1241 X:100
Cell: 615-426-7201
[Qr code  Description automatically generated]

From: Zeke Gabrielse ***@***.***>
Date: Monday, May 23, 2022 at 11:47 AM
To: keygen-sh/community ***@***.***>
Cc: James Felkins ***@***.***>, Author ***@***.***>
Subject: Re: [keygen-sh/community] Can not get algorithm aes-256-gcm+ed25519 for my license using java. (Discussion #29)

You need to encrypt the license file by setting the encrypt query parameter<https://keygen.sh/docs/api/licenses/#licenses-check-out-query-encrypt>. By default, license files are not encrypted.

GET https://api.keygen.sh/v1/accounts/{ACCOUNT}/licenses/{ID}/actions/check-out?encrypt=true

**Comment by felcodev** [comment]
So I thought I did, but maybe not. This is my url below.

https://api.keygen.sh/v1/accounts/0e43af21-f45a-4605-8a39-e2e05a66e1f5/licenses/a319998b-819e-487e-9c44-b8da349c3a02/actions/check-out

J.D. Felkins
FELCO Solutions, Inc.
www.felcosolutions.com<http://www.felcosolutions.com>
Email: ***@***.******@***.***>
Main: 770-766-1241 X:100
Cell: 615-426-7201
[Qr code  Description automatically generated]

From: Zeke Gabrielse ***@***.***>
Date: Monday, May 23, 2022 at 11:47 AM
To: keygen-sh/community ***@***.***>
Cc: James Felkins ***@***.***>, Author ***@***.***>
Subject: Re: [keygen-sh/community] Can not get algorithm aes-256-gcm+ed25519 for my license using java. (Discussion #29)

You need to encrypt the license file by setting the encrypt query parameter<https://keygen.sh/docs/api/licenses/#licenses-check-out-query-encrypt>. By default, license files are not encrypted.

GET https://api.keygen.sh/v1/accounts/{ACCOUNT}/licenses/{ID}/actions/check-out?encrypt=true

**Reply by ezekg** [reply]
You need to add `?encrypt=true` to the end of your check-out URL:

```
https://api.keygen.sh/v1/accounts/0e43af21-f45a-4605-8a39-e2e05a66e1f5/licenses/a319998b-819e-487e-9c44-b8da349c3a02/actions/check-out?encrypt=true
```
